src/results.rs

```diff
@@ -72,11 +79,39 @@ impl<'a> ExperimentResultDB for FileDB<'a> {
 
         Ok(())
     }
+
+    fn record_sha(&self, repo: &GitHubRepo, sha: &str) -> Result<()> {
+        // This avoids two threads writing on the same file together
+        let _lock = self.shafile_lock.lock().unwrap();
+
+        let mut existing = self.load_all_shas()?;
+        existing.insert(repo.clone(), sha.to_string());
+
+        let path = self.shafile_path();
+        if let Some(parent) = path.parent() {
+            fs::create_dir_all(parent)?;
+        }
+
+        let serializable: Vec<(GitHubRepo, String)> = existing.into_iter().collect();
```

Comment:
Why not just serialize the hashmap directly?

Reply:
Because it's not possible to have `GitHubRepo` as a key in json (it only allows strings). Sure, I could convert `GitHubRepo` to a string and then convert it back, but I prefer to have everything typed.